### PR TITLE
CI: Name specific steps in Travis CI matrix

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR uses the `name` feature for CI matrix steps.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,16 +4,18 @@ before_install:
...
+    - name: "jruby-9.1.8.0 on OpenJDK 8"
+      rvm: jruby-9.1.8.0
       jdk: openjdk8
     - rvm: jruby-9.2.8.0
-      env:
-        - JRUBY_OPTS="--debug"
+
+env:
+  global:
+    - JRUBY_OPTS="--debug"
```
Comment (dgmstuart) [inline .travis.yml]: This only applies to the JRUBY builds - but does this apply it to all builds? I guess it isn't harmful if it's present, but does it feel odd to be setting it for all of the MRI builds?

Comment (olleolleolle) [reply .travis.yml]: You are correct: This configuration sets it globally, to all matrix elements.

Most don't care about the presence of this environment variable.

The test suite is using `$JRUBY_OPTS` as a non-specific "just use the convenient --debug settings", as opposed to a matrix-selecting environment variable (e.g. ISOLATED=true, ISOLATED=false in the example at https://docs.travis-ci.com/user/build-matrix/ )

Comment (olleolleolle) [reply .travis.yml]: Would you like me to exclude this change from this PR?

Comment (dgmstuart) [reply .travis.yml]: @olleolleolle I'm not 100% understanding, but maybe it's simplest to exclude that change for now, if it's not directly related to the `name` change?

**Pull request merged**